### PR TITLE
v1.11 backports 2022-03-30

### PR DESCRIPTION
**Pull request by qmonnet**
* [ ] #18011 -- add context when return errors during datapath initialization (@kerthcet)
 * [x] #18336 -- bpf: Fix maglev hash with hostServices.hostNamespaceOnly (@ysksuzuki) :warning: Minor conflicts
 * [x] #18857 -- test: Flush CT tables after L7 proxy tests in K8sServices (@brb)
 * [x] #18869 -- test: Don't redeploy in AfterAll of K8sServices test case (@brb)
 * [x] #19104 -- Bpf fix conditional compilation (@jrajahalme)
 * [x] #19152 -- Fix log rotation of compressed logs (@chancez) :warning: Minor conflicts
 * [x] #19071 -- helm: Add RBAC Permissions to Clustermesh APIServer Clusterrole (@nathanjsweet)
 * [x] #19118 -- wireguard: Fix invalid bits when agent init (@Junnplus)
 * [ ] #19216 -- logo: fix position of central polygon (@sisp)
 * [x] #19228 -- workflows: Update call to Quay API (@pchaigno) :warning: Skipped files
 * [x] #19229 -- jenkinsfiles: Update calls to Quay API (@pchaigno)
 * [x] #19172 -- cmd: Fix issue where a ConfigMap value of `{}` was parsed as `map["{}":""]`. (@gandro)
 * [x] #19165 -- daemon: Bypass K8s cache check if K8s disabled (@christarazi)
 * [ ] #19193 -- pkg/redirectpolicy: Add missing namespace check in pod update handler (@aditighag)
 * [x] #18609 -- k8s: keep CiliumNode labels synced with Node object (@jibi) :warning: Minor conflicts (both commits), one file moved to `k8sT` instead of `k8s`
 * [ ] #19249 -- fix seg fault in logs (@Inode1)
 * [x] #19162 -- routing: Fix the incorrect deletion of IP rules (@pchaigno)
 * [x] #19042 -- clustermesh: Correct shared service annotation behaviour (@sayboras)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 18011 18336 18857 18869 19104 19152 19071 19118 19216 19228 19229 19172 19165 19193 18609 19249 19162 19042; do contrib/backporting/set-labels.py $pr done 1.11; done
```
or with
```
$ make add-label BRANCH=v1.11 ISSUES=18011,18336,18857,18869,19104,19152,19071,19118,19216,19228,19229,19172,19165,19193,18609,19249,19162,19042
```

### CODE REVIEW

**Review by sayboras — APPROVED** [review]
Thanks and looks good for my changes 💯

**Review by pchaigno — APPROVED** [review]
My changes look good. Thanks!

**Review by gandro — APPROVED** [review]
My changes and db3a9af7e49e55824dd8c58cc35a536d58199548 looks good to me. Thanks!

**Review by jibi — APPROVED** [review]
My PR looks good, thanks!

**Review by nathanjsweet — APPROVED** [review]
My changes look good. Thanks!

**Comment by qmonnet** [comment]
@jibi: I pushed again one of your commits, after renaming `test/k8s/Node.go` as `test/k8sT/Node.go` (note the `T`).

**Comment by qmonnet** [comment]
/test-backport-1.11

**Review by jibi — APPROVED** [review]
https://github.com/cilium/cilium/compare/d80b2bbb2c6e10c00cc5ad670c2c1810a53d5921..8fe003a8ab804e7d1d0909059e32524164dce0d5 looks good, approving again :heavy_check_mark: :heavy_check_mark:

**Review by christarazi — APPROVED** [review]

**Review by brb — APPROVED** [review]
My changes LGTM, thanks.

**Comment by qmonnet** [comment]
/test-1.23-4.9

**Comment by qmonnet** [comment]
`Cilium-PR-K8s-1.23-kernel-4.9` passed this time. Marking as ready-to-merge.

File: test/k8sT/Services.go [hunk 1]
```diff
@@ -254,6 +260,9 @@ var _ = SkipDescribeIf(helpers.RunsOn54Kernel, "K8sServicesTest", func() {
 			BeforeAll(func() {
 				DeployCiliumOptionsAndDNS(kubectl, ciliumFilename, map[string]string{
 					"hostServices.hostNamespaceOnly": "true",
+					// Enable Maglev to check if the Maglev LUT for ClusterIP is properly populated,
+					// and external clients can access ClusterIP with it.
+					"loadBalancer.algorithm": "maglev",
```
Comment (pchaigno) [inline test/k8sT/Services.go:265]: @qmonnet I guess the minor conflict for https://github.com/cilium/cilium/pull/18336 's backport was on this file. Any reason the [first few lines](https://github.com/cilium/cilium/pull/18336/commits/2769f6f647df6b6e66214963b483ab5df4c581b8#diff-726ed9ddc3bc0d4430eaa9911b2496376f0f5a8445ff89988966eb216328ca2aR459) weren't backported? As far as I can see, https://github.com/cilium/cilium/blob/8fe003a8ab804e7d1d0909059e32524164dce0d5/test/k8sT/Services.go#L754 should be changed.

Comment (qmonnet) [reply test/k8sT/Services.go:265]:
```
[ Backport notes: Minor conflicts in test/k8sT/Services.go; the second
    hunk in that file was skipped (the redeploy is not present in this
    branch). ]
```

But you're correct, I missed it when doing the backport. Given that [we backported the restructuring of the tests](https://github.com/cilium/cilium/commit/298fb49a4951324b1f14ac0b61dc1a1f09d74705), this one was not where I expected it to be. I'll fix it and re-push. Thanks!

**Comment by qmonnet** [comment]
/test-only --focus="K8sServicesTest.*With ClusterIP external access"

**Review by pchaigno — APPROVED** [review]
Backport of https://github.com/cilium/cilium/pull/18336 looks good to me.

**Comment by pchaigno** [comment]
/test-backport-1.11

**Review by jrajahalme — APPROVED** [review]
Looks good on my part :-)

**Review by chancez — APPROVED** [review]
Backport of https://github.com/cilium/cilium/pull/19152 7342c4bfb1e880fdb548b81c671fcb472cae9677 looks good to me

**Comment by pchaigno** [comment]
All backports with conflcits have been reviewed. All required tests are passing except for Runtime which is broken on all stable branches (cf. https://github.com/cilium/cilium/issues/18919). A bunch of other tests are failing but that's because I mistakenly triggered the master tests with `/test`. Merging.

**Pull request merged**